common/console.cpp

```diff
@@ -111,7 +112,16 @@ namespace console {
             }
         }
 
-        setlocale(LC_ALL, "");
+        auto locale = setlocale(LC_ALL, "");
+
+        if (locale == nullptr || strcasestr(locale, "utf-8") == nullptr) {
+            auto lang = getenv("LANG");
+            if (lang != nullptr && strcasestr(lang, "utf-8") != nullptr) {
+                setlocale(LC_ALL, lang);
+            } else {
+                setlocale(LC_ALL, "C.UTF-8");
+            }
+        }
```

Comment:
If we're trying to cover all our bases, we probably want to try locales in this order: "", "LANG", "C.UTF-8", "C.utf8".

Unfortunately, this can get a bit complicated, but if we approach it with a loop, I think it looks the cleanest and lets us add additional sets in the future. Like perhaps we want to check `en_US.UTF-8` if the other UTF 8 character sets fail to load since not all distros include the C UTF 8 set, or perhaps there's another function someone will recommend/write that will suggest a possible locale.

```suggestion
        using LocaleStr = const char*;
        LocaleStr locales[] = {
            "", 
            getenv("LANG"), 
            "C.UTF-8", 
            "C.utf8"
        };
        const int numLocales = sizeof(locales) / sizeof(locales[0]);

        for (int i = 0; i < numLocales; ++i) {
            LocaleStr current = locales[i];
            if (current) {
                LocaleStr result = setlocale(LC_ALL, current);
                if (result && (strcasestr(result, "utf-8") || strcasestr(result, "utf8"))) {
                    break;
                }
            }
        }
```

I realize this is slightly less efficient, but it will cover the widest range of character sets.

Reply:
Since setting incorrect/nonexistent locale will simply make `setlocale` return nullptr, so I believe this could be simplified to 
`setlocale(LC_ALL, "C.UTF-8") || setlocale(LC_ALL, "C.utf8") || setlocale(LC_ALL, "");` in the else bracket

Reply:
That would *probably* work. We don't use any of the locale formatting functions or locale dependent functions such as `isalpha`. But, we aren't the only ones adding code and although I don't see the need for such uses, it is possible someone wants to format the timing results or benchmarks in the future.

I think you had the right idea starting with `setlocale(LC_ALL, "")` and checking what it returned for UTF 8. And if you want to try `"LANG"` (which I believe is traditionally used as a fallback when the default fails), then we also need to test if that locale is UTF 8. But for the last two, the simpler check `setlocale(LC_ALL, "C.UTF-8") || setlocale(LC_ALL, "C.utf8")` works, being that the utf8 test is redundant.

The other thing I thought of later was if `setlocale(LC_ALL, "")` returned a string that didn't contain utf8, we could truncate it after the `.` and try it with `utf8` and `UTF-8`. I don't know how elaborate we want to get. I googled it and C.utf8 C.UTF-8 isn't included in all distros by default, but all the systems I checked did have it. For example:
```bash
$ locale -a
C
C.utf8
en_US.utf8
POSIX
```

Reply:
> The other thing I thought of later was if `setlocale(LC_ALL, "")` returned a string that didn't contain utf8, we could truncate it after the `.` and try it with `utf8` and `UTF-8`. I don't know how elaborate we want to get. I googled it and C.utf8 C.UTF-8 isn't included in all distros by default, but all the systems I checked did have it.

Apparently there isn't a *hard* standard for locale (though obviously vast majority of Linux distros use the same "standard" )

GNU libc manual says " locale names are system-specific. "

So no matter what we do, this is always gonna be a best effort workaround and never a "proper" fix.

I believe that including `setlocale(LC_ALL, "")` as a final fallback, would be sufficient to appropriately "cancel" this fix when it cannot succeed, so the previous behaviour will be kept if the workaround cannot be applied, and this will at least ensure this doesn't accidentally make things worse in case of a weird system configuration.

Reply:
I've been sitting on a big patch that adds some much needed functionality to console based editing until I had enough time to properly address any issues it may raise. I could try a more elaborate fix in that patch since it will need to be tested more thoroughly anyway and we could go with your much simpler fix in the meantime. I can't imagine using `C` instead of `language[_territory]` would create any issues, but I haven't tested it. If you have done so with Polish and everything looked fine, I'll approve that change. We just need to keep an eye out for any new reports that might be related.